Add files via upload

antonychiu2 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

antonychiu2 committed:
Add files via upload

Commit:
Auto commit for SQL_Injection-19892808-31bf-41bd-83bd-acbbb95e762f

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/19892808-31bf-41bd-83bd-acbbb95e762f)

Commit:
Auto commit for SQL_Injection-3d578d78-0ce3-4ca9-af80-52882e196c80

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/3d578d78-0ce3-4ca9-af80-52882e196c80)

Commit:
Auto commit for SQL_Injection-31f59824-3e15-411c-a792-81d10b0d2f05

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/31f59824-3e15-411c-a792-81d10b0d2f05)

Commit:
Auto commit for SQL_Injection-f961c93c-ab82-4c9f-ba75-cac588a7d8a6

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/f961c93c-ab82-4c9f-ba75-cac588a7d8a6)

Commit:
Auto commit for SQL_Injection-61324ccb-5185-4421-a5a3-034e67d8e33d

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/61324ccb-5185-4421-a5a3-034e67d8e33d)

Commit:
Auto commit for SQL_Injection-3789b19e-6203-451e-b7d1-98ed31144572

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/3789b19e-6203-451e-b7d1-98ed31144572)

Commit:
Auto commit for SQL_Injection-e98c96cc-488a-451a-b941-294d1952c709

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/e98c96cc-488a-451a-b941-294d1952c709)

Commit:
Auto commit for SQL_Injection-0810d0a4-55e4-4a76-90cd-b8a204884dd3

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/0810d0a4-55e4-4a76-90cd-b8a204884dd3)

Commit:
Auto commit for SQL_Injection-af924ecd-dfcc-443c-a3be-a486d0ac6e0c

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/af924ecd-dfcc-443c-a3be-a486d0ac6e0c)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -31,6 +31,6 @@
 module.exports = {
     CertGraphModel: certGraphModel,
     getGraphDataByZone: function (zone) {
-        return certGraphModel.findOne({ 'zone': zone }).exec();
+        return certGraphModel.findOne({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/0810d0a4-55e4-4a76-90cd-b8a204884dd3)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -38,6 +38,6 @@
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
-        return (graphDataModel.countDocuments({ 'zone': zone }).exec());
+        return (graphDataModel.countDocuments({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec());
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/e98c96cc-488a-451a-b941-294d1952c709)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -45,7 +45,7 @@
         return (promise);
     },
     getTPDsByTPD: function (tpd) {
-        return tpdModel.findOne({ 'tld': tpd });
+        return tpdModel.findOne({ 'tld': mongoSanitize.sanitize({ data: tpd }).data });
     },
     getTPDsByWildcard: function (search, listOnly) {
         let promise;
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/3d578d78-0ce3-4ca9-af80-52882e196c80)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -52,7 +52,7 @@
         let AWSregex = new RegExp('.*' + search + '$');
         if (listOnly) {
             promise = tpdModel.find({
-                'tld': { '$regex': AWSregex },
+                'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
             promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/f961c93c-ab82-4c9f-ba75-cac588a7d8a6)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -31,7 +31,7 @@
     getGraphDocsByZone: function (zone) {
         let limitQuery = { 'docs': 1 };
         return graphDocsModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/61324ccb-5185-4421-a5a3-034e67d8e33d)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -36,7 +36,7 @@
     getTPDsByZone: function (zone, listOnly) {
         let promise;
         if (!listOnly) {
-            promise = tpdModel.find({ 'zones.zone': zone }).exec();
+            promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
                 'zones.zone': zone,
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/19892808-31bf-41bd-83bd-acbbb95e762f)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -34,7 +34,7 @@
     getGraphLinksByZone: function (zone) {
         let limitQuery = { 'links': 1 };
         return graphLinksModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/3789b19e-6203-451e-b7d1-98ed31144572)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -34,7 +34,7 @@
     getGraphConfigByZone: function (zone) {
         let limitQuery = { 'config': 1 };
         return graphModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/af924ecd-dfcc-443c-a3be-a486d0ac6e0c)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -34,7 +34,7 @@
     getGraphDataByZone: function (zone) {
         let limitQuery = { 'data': 1, 'errs': 1 };
         return graphDataModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/771c5980-b9b9-4b78-8ab1-dca795437475/fix/31f59824-3e15-411c-a792-81d10b0d2f05)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Commit:
Auto commit for SQL_Injection-395b82d7-e27c-43cc-9e3f-441ca7285340

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/395b82d7-e27c-43cc-9e3f-441ca7285340)

Commit:
Auto commit for SQL_Injection-b0317c85-71fe-4792-bc83-c7d85455dc45

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/b0317c85-71fe-4792-bc83-c7d85455dc45)

Commit:
Auto commit for SQL_Injection-2f0cc009-eb44-4b1d-8252-929d4f8af13f

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/2f0cc009-eb44-4b1d-8252-929d4f8af13f)

Commit:
Auto commit for SQL_Injection-7070bf07-aba1-47ed-be04-798aba8af557

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/7070bf07-aba1-47ed-be04-798aba8af557)

Commit:
Auto commit for SQL_Injection-435123f6-454d-4830-8cf3-c0e57e30a879

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/435123f6-454d-4830-8cf3-c0e57e30a879)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphModel: graphModel,
     getGraphConfigByZone: function (zone) {
-        let limitQuery = { 'config': 1 };
+        let limitQuery = { 'config': mongoSanitize.sanitize({ data: 1 }).data };
         return graphModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/435123f6-454d-4830-8cf3-c0e57e30a879)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphLinksModel: graphLinksModel,
     getGraphLinksByZone: function (zone) {
-        let limitQuery = { 'links': 1 };
+        let limitQuery = { 'links': mongoSanitize.sanitize({ data: 1 }).data };
         return graphLinksModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/b0317c85-71fe-4792-bc83-c7d85455dc45)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -40,7 +40,7 @@
             promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
-                'zones.zone': zone,
+                'zones.zone': mongoSanitize.sanitize({ data: zone }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         }
         return (promise);
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/2f0cc009-eb44-4b1d-8252-929d4f8af13f)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -56,7 +56,7 @@
                 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
-            promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
+            promise = tpdModel.find({ 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data }).exec();
         }
         return (promise);
     },
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/7070bf07-aba1-47ed-be04-798aba8af557)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -30,7 +30,7 @@
 module.exports = {
     GraphDocsModel: graphDocsModel,
     getGraphDocsByZone: function (zone) {
-        let limitQuery = { 'docs': 1 };
+        let limitQuery = { 'docs': mongoSanitize.sanitize({ data: 1 }).data };
         return graphDocsModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/d07ea6d9-d042-488e-ba3f-a00c999feda4/fix/395b82d7-e27c-43cc-9e3f-441ca7285340)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.
